scripts/eosio_build_ubuntu.sh

```diff
@@ -155,102 +162,198 @@ if [ $? -ne 0 ]; then exit -1; fi
+      && mkdir extra && cd extra \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/clang-tools-extra.git \
+      && cd clang-tools-extra && git checkout $PINNED_COMPILER_CLANG_TOOLS_EXTRA_COMMIT && cd .. \
+      && cd ../../../../projects \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/libcxx.git \
+      && cd libcxx && git checkout $PINNED_COMPILER_LIBCXX_COMMIT && cd ../ \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/libcxxabi.git \
+      && cd libcxxabi && git checkout $PINNED_COMPILER_LIBCXXABI_COMMIT && cd ../ \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/libunwind.git \
+      && cd libunwind && git checkout $PINNED_COMPILER_LIBUNWIND_COMMIT && cd ../ \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/compiler-rt.git \
+      && cd compiler-rt && git checkout $PINNED_COMPILER_COMPILER_RT_COMMIT && cd ../ \
+      && cd ${TMP_LOCATION}/clang8 \
+      && mkdir build && cd build \
+      && $CMAKE -G "Unix Makefiles" -DCMAKE_INSTALL_PREFIX="${CLANG8_ROOT}" -DLLVM_BUILD_EXTERNAL_COMPILER_RT=ON -DLLVM_BUILD_LLVM_DYLIB=ON -DLLVM_ENABLE_LIBCXX=ON -DLLVM_ENABLE_RTTI=ON -DLLVM_INCLUDE_DOCS=OFF -DLLVM_OPTIMIZED_TABLEGEN=ON -DLLVM_TARGETS_TO_BUILD=host -DCMAKE_BUILD_TYPE=Release .. \
+      && make -j"${JOBS}" \
+      && make install \
+      && rm -rf ${TMP_LOCATION}/clang8 \
+      && cd ../.. \
+      || exit 1
 
-printf "Checking Boost library (${BOOST_VERSION}) installation...\\n"
-BOOSTVERSION=$( grep "#define BOOST_VERSION" "$BOOST_ROOT/include/boost/version.hpp" 2>/dev/null | tail -1 | tr -s ' ' | cut -d\  -f3 )
-if [ "${BOOSTVERSION}" != "${BOOST_VERSION_MAJOR}0${BOOST_VERSION_MINOR}0${BOOST_VERSION_PATCH}" ]; then
-	printf "Installing Boost library...\\n"
-	curl -LO https://dl.bintray.com/boostorg/release/${BOOST_VERSION_MAJOR}.${BOOST_VERSION_MINOR}.${BOOST_VERSION_PATCH}/source/boost_$BOOST_VERSION.tar.bz2 \
-	&& tar -xjf boost_$BOOST_VERSION.tar.bz2 \
-	&& cd $BOOST_ROOT \
-	&& ./bootstrap.sh --prefix=$BOOST_ROOT \
-	&& ./b2 -q -j"${JOBS}" install \
-	&& cd .. \
-	&& rm -f boost_$BOOST_VERSION.tar.bz2 \
-	&& rm -rf $BOOST_LINK_LOCATION \
-	&& ln -s $BOOST_ROOT $BOOST_LINK_LOCATION \
-	|| exit 1
-	printf " - Boost library successfully installed @ ${BOOST_ROOT} (Symlinked to ${BOOST_LINK_LOCATION}).\\n"
-else
-	printf " - Boost library found with correct version @ ${BOOST_ROOT} (Symlinked to ${BOOST_LINK_LOCATION}).\\n"
-fi
-if [ $? -ne 0 ]; then exit -1; fi
+      printf " - Clang 8 successfully installed @ ${CLANG8_ROOT}\\n"
+   else
+      printf " - Clang 8 found @ ${CLANG8_ROOT}.\\n"
+   fi
+   if [ $? -ne 0 ]; then exit -1; fi
 
+   printf "\\n"
 
-printf "\\n"
+   printf "Checking LLVM 4 installation...\\n"
+   if [ ! -d $OPT_LOCATION/llvm4 ] || [ $FORCE_BUILD ]; then
```

Comment:
don't we only want to build this when not pinning compiler? otherwise it won't work right -- because it's using the toolchain file

Reply:
n/m I see it's good

Reply:
Huh? We only want to build when pinning because we are using a toolchain file.  We are pinning to libcxx for all of pinned builds.